API/controllers/userController.js

```diff
@@ -0,0 +1,97 @@
+const express = require('express');
+const mongoose = require('mongoose');
+const jwt = require('jsonwebtoken');
+const {jwtSecret} = require('../config/secrets');
+const {verifyToken} = require('../helpers/authorization');
+const router = express.Router();
+
+require('../models/User');
+const User = mongoose.model('User');
+
+// Return all users
+router.get('/', verifyToken, (req, res) => {
+	User.find().then(users => {
+		users.forEach(user => {
+			delete user.password;
+		});
+		res.json(users);
+	});
+});
+
+// Get specific user
+router.get('/user/:id', (req, res) => {
+	User.findOne({_id: req.params.id}).then(user => {
+		delete user.password;
+		res.json(user);
+	}).catch(err => {
+		res.json(err);
+	});
+});
+
+// Creates a user -- Protect this route with `golden ticket`
+router.post('/', (req, res) => {
+	User.create(req.body).then(user => {
+		res.json(user);
+	}).catch(err => {
+		res.json(err);
+	});
+});
+
+// Deletes a user
+router.delete('/:id', verifyToken, (req, res) => {
+	if(req.params.id !== req.user._id) {
+		res.status(403);
+	}
+	else {
+		User.deleteOne({_id: req.params.id})
+		.then(() => {
+			res.status(200).send('success');
+		}).catch(err => {
+			res.status(404).json(err);
+		});
+	}
+});
+
+// Updates logged in user
+router.put('/loggedInUser', verifyToken, (req, res) => {
```

Comment:
So if I understand it correctly, I think having the /:id route like in the delete method would just add better error handling for the transaction. The only reason `req.params.id` wouldn't be the same as the token-provided id is if the client is trying to update the 'wrong' user (any other id) - instead of going ahead and updating the logged in user (not what the client explicitly intended), it's probably better to yell at them with a 403 so they know they got the id wrong.
I think the extra request to `loggedInUser` probably wouldn't hurt, it's not like people will be updating / deleting their accounts often.

Reply:
Yeah that pretty much sums up how I felt, just wanted to offer both options but I think I'll change it back 😄